modules/geopackage/src/lib/parse-geopackage.js

```diff
@@ -381,3 +414,54 @@ function interleaveResults(columns, values) {
 
   return merged;
 }
+
+/**
+ * Get arrow schema
+ *
+ * @param  {Database} db GeoPackage object
+ * @param  {string} tableName  table name
+ * @return {Schema | null} Arrow-like Schema
+ */
+function getArrowSchema(db, tableName) {
+  const metadata = db.exec(
+    `SELECT * FROM \`sqlite_master\` WHERE type='table' AND tbl_name='${tableName}';`
+  )[0];
```

Comment:
This query will be called on every table that exists in the GeoPackage. I haven't benchmarked, but I assume there's a non-negligible constant overhead to each SQL query.

Instead, I'd suggest doing a single query to retrieve the entire `sqlite_master` table (presumably it isn't much data) and then passing that result into each `getVectorTable` call. This would be similar to what I do for the projections table. I load the full set of projections _once_ and then pass that as an object into each `getVectorTable` call

https://github.com/visgl/loaders.gl/blob/97edd7ae1b236e8962c15dcfb5cedcfd1206aa07/modules/geopackage/src/lib/parse-geopackage.js#L60

https://github.com/visgl/loaders.gl/blob/97edd7ae1b236e8962c15dcfb5cedcfd1206aa07/modules/geopackage/src/lib/parse-geopackage.js#L66

Reply:
Also, are we sure this is the simplest way to get type information about each column? I see that when you query `sqlite_master` you get the SQL string that would recreate the given table.

Why not use
```sql
SELECT name, type FROM pragma_table_info('<table_name>')
```

Then as output you get a simple mapping from column name to SQL type. No complex regex required.

```
+------------+----------+
| name       | type     |
+------------+----------+
| id         | INTEGER  |
| geom       | GEOMETRY |
| property_0 | TEXT     |
| property_1 | TEXT     |
| property_2 | TEXT     |
+------------+----------+
```

Reply:
Done